src/parser/mod.rs

```diff
@@ -3832,9 +3811,11 @@ impl<'a> Parser<'a> {
 
     /// If the current token is the `expected` keyword, consume the token.
     /// Otherwise, return an error.
+    ///
+    // todo deprecate infavor of expected_keyword_is
     pub fn expect_keyword(&mut self, expected: Keyword) -> Result<TokenWithSpan, ParserError> {
-        if let Some(token) = self.parse_keyword_token_ref(expected) {
-            Ok(token.clone())
+        if self.parse_keyword(expected) {
```

Comment:
I actually think using `parse_keyword` and `get_current_token` is clearer than the original formulation of `paser_keyword_token_ref` and checking for `Some`